contextual_lenses/pfam_family_classification/pfam_one_hot_lenses_parallel.py

```diff
@@ -0,0 +1,223 @@
+        with open(os.path.join(data_dir, partition, fn)) as f:
+            shards.append(pd.read_csv(f, index_col=None))
+    return pd.concat(shards)
+
+test_df = read_all_shards('test')
+dev_df = read_all_shards('dev')
+train_df = read_all_shards('train')
+
+partitions = {'test': test_df, 'dev': dev_df, 'train': train_df}
+for name, df in partitions.items():
+    print('Dataset partition "%s" has %d sequences' % (name, len(df)))
+
+def mod_family_accession(family_accession):
+  return family_accession[:family_accession.index('.')]
+
+train_df['mod_family_accession'] = train_df.family_accession.apply(lambda x: mod_family_accession(x))
+test_df['mod_family_accession'] = test_df.family_accession.apply(lambda x: mod_family_accession(x))
+dev_df['mod_family_accession'] = dev_df.family_accession.apply(lambda x: mod_family_accession(x))
+
+kept_family_accessions = []
+for i in range(1, 101):
+  kept_family_accessions.append('PF00' + '0'*(3-len(str(i))) + str(i))
+
+def keep_family_accessions(df, kept_family_accessions):
+  
+  dfs_by_family_accessions = []
+  
+  for family_accession in kept_family_accessions:
+    df_by_family_accession = df[df['mod_family_accession']==family_accession]
+    dfs_by_family_accessions.append(df_by_family_accession)
+  
+  return pd.concat(dfs_by_family_accessions)
+
+train_df = keep_family_accessions(train_df, kept_family_accessions)
+test_df = keep_family_accessions(test_df, kept_family_accessions)
+dev_df = keep_family_accessions(dev_df, kept_family_accessions)
+
+family_ids = sorted(set(train_df['family_id'].values))
+family_id_to_index = {}
+for i, family_id in enumerate(family_ids):
+  family_id_to_index[family_id] = i
+
+train_df['index'] = train_df.family_id.apply(lambda x: family_id_to_index[x])
+test_df['index'] = test_df.family_id.apply(lambda x: family_id_to_index[x])
+dev_df['index'] = dev_df.family_id.apply(lambda x: family_id_to_index[x])
+
+test_indexes = test_df['index'].values
+
+num_families = len(set(train_df['family_id']))
+
+def pad_seq(seq, pad_char='-'):
+  SEQ_LEN = 512
+  seq = seq[:SEQ_LEN]
+  padded_seq = seq + pad_char*(SEQ_LEN-len(seq))
+  return padded_seq
+
+AMINO_ACID_VOCABULARY = [
+    'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R',
+    'S', 'T', 'V', 'W', 'Y', 'X', 'U', 'B', 'O', 'Z', '-'
+]
+num_categories = len(AMINO_ACID_VOCABULARY)
+
+def residues_to_one_hot_inds(amino_acid_residues):
+  return np.array([AMINO_ACID_VOCABULARY.index(char) for char in amino_acid_residues])
+
+train_df['one_hot_inds'] = train_df.sequence.apply(lambda x: residues_to_one_hot_inds(pad_seq(x)))
+test_df['one_hot_inds'] = test_df.sequence.apply(lambda x: residues_to_one_hot_inds(pad_seq(x)))
+dev_df['one_hot_inds'] = dev_df.sequence.apply(lambda x: residues_to_one_hot_inds(pad_seq(x)))
+
+print(train_df.head())
+
+
+"""Data batching."""
+def create_data_iterator(df, batch_size, epochs=1, buffer_size=None, seed=0, drop_remainder=False):
```

Comment:
as you said earlier, this should be merged with the create_data_iterator function for gfp. Just input df and the name of the columns to use for inputs and outputs. Necessary for the PR, but good to do in medium term.